test/e2e/enveloped_object_placement_test.go

```diff
@@ -0,0 +1,309 @@
+		}
+		return nil
+	}
+}
+
+func checkForOneClusterFailedToApplyStatus(wantSelectedResources []placementv1beta1.ResourceIdentifier) func() error {
+	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+	workNamespaceName := fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess())
+	failedResourcePlacement := []placementv1beta1.FailedResourcePlacement{
+		{
+			ResourceIdentifier: placementv1beta1.ResourceIdentifier{
+				Kind:      "ResourceQuota",
+				Name:      testEnvelopeResourceQuota.Name,
+				Version:   "v1",
+				Namespace: testEnvelopeResourceQuota.Namespace,
+				Envelope: &placementv1beta1.EnvelopeIdentifier{
+					Name:      testEnvelopConfigMap.Name,
+					Namespace: workNamespaceName,
+					Type:      placementv1beta1.ConfigMapEnvelopeType,
+				},
+			},
+			Condition: metav1.Condition{
+				Type:   string(placementv1beta1.WorkConditionTypeApplied),
+				Status: metav1.ConditionFalse,
+				Reason: work.AppliedManifestFailedReason,
+			},
+		},
+	}
+
+	return func() error {
+		crp := &placementv1beta1.ClusterResourcePlacement{}
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+			return err
+		}
+		wantCRPConditions := crpSyncFailedConditions(crp.Generation)
+		if diff := cmp.Diff(crp.Status.Conditions, wantCRPConditions, crpStatusCmpOptions...); diff != "" {
+			return fmt.Errorf("CRP status diff (-got, +want): %s", diff)
+		}
+		// check the selected resources is still right
+		if diff := cmp.Diff(crp.Status.SelectedResources, wantSelectedResources, crpStatusCmpOptions...); diff != "" {
+			return fmt.Errorf("CRP status diff (-got, +want): %s", diff)
+		}
+		// check the placement status has a failed placement
+		applyFailed := false
+		for _, placementStatus := range crp.Status.PlacementStatuses {
+			if len(placementStatus.FailedResourcePlacements) != 0 {
+				applyFailed = true
+			}
+		}
+		if !applyFailed {
+			return fmt.Errorf("CRP status does not have failed placement")
+		}
+		for _, placementStatus := range crp.Status.PlacementStatuses {
+			// this is the cluster that got the new enveloped resource that was malformed
+			if len(placementStatus.FailedResourcePlacements) != 0 {
+				if diff := cmp.Diff(placementStatus.FailedResourcePlacements, failedResourcePlacement, crpStatusCmpOptions...); diff != "" {
+					return fmt.Errorf("CRP status diff (-got, +want): %s", diff)
+				}
+				if !strings.Contains(placementStatus.FailedResourcePlacements[0].Condition.Message, "unsupported scope") {
+					return fmt.Errorf("CRP failed resource placement does not have unsupported scope message")
+				}
+				if diff := cmp.Diff(placementStatus.Conditions, resourcePlacementApplyFailedConditions(crp.Generation), crpStatusCmpOptions...); diff != "" {
+					return fmt.Errorf("CRP status diff (-got, +want): %s", diff)
+				}
+			} else {
+				// the cluster is stuck behind a rollout schedule since we now have 1 cluster that is not in applied ready status
```

Comment:
should we wait until these clusters are in the stable state then compare the status?

We may stop validating if hit this "else" condition and may miss the regression.

Reply:
this is the stable state. What else can happen?